rmw_zenoh_cpp/src/detail/rmw_context_impl_s.cpp

```diff
@@ -35,370 +37,502 @@
 // TODO(clalancette): Make this configurable, or get it from the configuration
 #define SHM_BUFFER_SIZE_MB 10
 
-///=============================================================================
-void rmw_context_impl_s::graph_sub_data_handler(const z_sample_t * sample, void * data)
+// This global mapping of raw Data pointers to Data shared pointers allows graph_sub_data_handler()
+// to lookup the pointer, and gain a reference to a shared_ptr if it exists.
+// This guarantees that the Data object will not be destroyed while we are using it.
+static std::mutex data_to_data_shared_ptr_map_mutex;
+static std::unordered_map<Data *, std::shared_ptr<Data>> data_to_data_shared_ptr_map;
```

Comment:
include `<memory>`

Reply:
Fixed in dcd2e7fab758d23e550adc1c1167800b45c08594